src/components/apps/admin/referenceGenomes/ReferenceGenomes.js

```diff
@@ -0,0 +1,195 @@
+                });
+                queryCache.invalidateQueries(ADMIN_REFERENCE_GENOMES_QUERY_KEY);
+            },
+            onError: (e) => {
+                showErrorAnnouncer(t("createFailed"), e);
+            },
+        }
+    );
+
+    const dataMemo = React.useMemo(() => data?.genomes, [data]);
+
+    const columns = React.useMemo(
+        () => [
+            {
+                Header: t("name"),
+                accessor: refGenomeConstants.keys.NAME,
+                Cell: ({ row, value }) => (
+                    <>
+                        {row?.original?.deleted && (
+                            <RemoveCircleIcon
+                                style={{ color: theme.palette.error.main }}
+                            />
+                        )}
+                        <Link
+                            style={{ cursor: "pointer" }}
+                            onClick={() => {
+                                setSelectedReferenceGenome(row.original);
+                            }}
+                        >
+                            {value}
+                        </Link>
+                    </>
+                ),
+            },
+            {
+                Header: t("path"),
+                accessor: refGenomeConstants.keys.PATH,
+            },
+            {
+                Header: t("createdBy"),
+                accessor: refGenomeConstants.keys.CREATED_BY,
+            },
+            {
+                Header: t("createdOn"),
+                accessor: refGenomeConstants.keys.CREATED_ON,
+            },
+        ],
+        [t, theme.palette.error.main]
+    );
+
+    useEffect(() => {
+        if (selectedReferenceGenome) {
+            setEditDialogOpen(true);
+        }
+    }, [selectedReferenceGenome]);
+
+    useEffect(() => {
+        if (error) {
+            const errorString = JSON.stringify(error);
+            router.push(
+                `/${NavigationConstants.ERROR}?errorInfo=${errorString}`
+            );
+        }
+    }, [error, router]);
+
+    if (isFetching || genomeMutationStatus || genomeCreationStatus || !data) {
+        return <TableLoading numColumns={5} numRows={100} baseId={baseId} />;
+    }
+
+    return (
+        <>
+            <EnhancedTable
+                columns={columns}
+                data={dataMemo}
+                baseId={baseId}
+                onAddClicked={() => {
+                    const user = userProfile?.id;
+                    const currentDate = formatDateObject(
+                        new Date(),
+                        dateConstants.LONG_DATE_FORMAT
+                    );
+                    setSelectedReferenceGenome({
+                        created_by: user,
+                        created_on: currentDate,
+                        last_modified_on: currentDate,
+                        last_modified_by: user,
+                    });
+                    setEditDialogOpen(true);
```

Comment:
Seems like this line isn't needed due to the `useEffect` or vice versa - the `useEffect` is not necessary